tests/unit_tests/client/address_book/impl.cc

```diff
@@ -240,6 +240,54 @@ BOOST_AUTO_TEST_CASE(PGPClearSign)
   BOOST_CHECK(Validate());
 }
 
+BOOST_AUTO_TEST_CASE(InsertTooShortBase64)
+{
+  std::string address{"kovri.i2p"}, base64{"(_!@*Z"};
+  BOOST_CHECK_THROW(
+      book.InsertAddressIntoStorage(address, base64), std::length_error);
+}
+
+BOOST_AUTO_TEST_CASE(InsertNoStorageFile)
+{
+  std::string address{"anonimal.i2p"},
+      base64{
+          "AQZGLAMpI9Q0l0kmMj1vpJJYK3CjLp~fE3MfvE-"
+          "e7KMKjI5cPOH6EN8m794uHJ6b09qM8mb9VEv1lVLEov~"
+          "usVliTSXCSHuRBOCIwIOuDNU0AbVa4BpIx~"
+          "2sU4TxKhoaA3zQ6VzINoduTdR2IJhPvI5xzezp7dR21CEQGGTbenDslXeQ4iLHFA2~"
+          "bzp1f7etSl9T2W9RID-"
+          "KH78sRQmzWnv7dbhNodMbpO6xsf1vENf6bMRzqD5vgHEHZu2aSoNuPyYxDU1eM6--"
+          "61b2xp9mt1k3ud-5WvPVg89RaU9ugU5cxaHgR927lHMCAEU2Ax~"
+          "zUb3DbrvgQBOTHnJEx2Fp7pOK~"
+          "PnP6ylkYKQMfLROosLDXinxOoSKP0UYCh2WgIUPwE7WzJH3PiJVF0~WZ1dZ9mg00c~"
+          "gzLgmkOxe1NpFRNg6XzoARivNVB5NuWqNxr5WKWMLBGQ9YHvHO1OHhUJTowb9X90BhtH"
+          "nLK2"
+          "AHwO6fV-iHWxRJyDabhSMj1kuYpVUBQAEAAcAAA=="};
+  BOOST_CHECK_THROW(
+      book.InsertAddressIntoStorage(address, base64), std::runtime_error);
+}
+
+BOOST_AUTO_TEST_CASE(InsertMalformedBase64)
+{
+  std::string address{"anonimal.i2p"},
+      base64{
+          "AQZGLAMpI9Q0l0kmMj1vpJJYK3CjLp~fE3MfvE-"
+          "e7KMKjI5cPOH6EN8m794uHJ6b09qM8mb9VEv1lVLEov~"
+          "usVliTSXCSHuRBOCIwIOuDNU0AbVa4BpIx~"
+          "2sU4TxKhoaA3zQ6VzINoduTdR2IJhPvI5xzezp7dR21CEQGGTbenDslXeQ4iLHFA2~"
+          // Invalid base64 chunk
+          "{{{{{{{{{{{{{{{{{{{"
+          "KH78sRQmzWnv7dbhNodMbpO6xsf1vENf6bMRzqD5vgHEHZu2aSoNuPyYxDU1eM6--"
+          "61b2xp9mt1k3ud-5WvPVg89RaU9ugU5cxaHgR927lHMCAEU2Ax~"
+          "zUb3DbrvgQBOTHnJEx2Fp7pOK~"
+          "PnP6ylkYKQMfLROosLDXinxOoSKP0UYCh2WgIUPwE7WzJH3PiJVF0~WZ1dZ9mg00c~"
+          "gzLgmkOxe1NpFRNg6XzoARivNVB5NuWqNxr5WKWMLBGQ9YHvHO1OHhUJTowb9X90BhtH"
+          "nLK2"
+          "AHwO6fV-iHWxRJyDabhSMj1kuYpVUBQAEAAcAAA=="};
+  BOOST_CHECK_THROW(
+      book.InsertAddressIntoStorage(address, base64), std::runtime_error);
+}
```

Comment:
These tests are currently useless because:

  - You're intentionally creating the filesystem failure within the implementation by ignoring the fact that a file doesn't exist. This test-case will always be hopeless because it **should not rely on the filesystem** as I have [already said](https://github.com/monero-project/kovri/pull/813#issuecomment-367224433).
  - You're simply testing our decoder for which we already have tests.

Other issues:

- Using curly-braces to initialize a string-literal ctor?! The literal is not a list nor aggregate nor integral. Use `()` regardless of what the standard allows. Doing so is consistent with rest of this codebase and nearly every other c++ program out there.
- Use const correctness.
- Keep the base64 string on one line. No one wants to see a line split up like that and clang-format isn't bright when it comes to this issue. This is not your fault though because the guide clearly says to run clang-format. We should make an addendum to the guide.

Reply:
> These tests are currently useless because

I'll remove these tests, and try to come up with new cases.

> Keep the base64 string on one line. No one wants to see a line split up like that and clang-format isn't bright when it comes to this issue.

Noted. I've turned off auto-clang-formatting on save to hopefully cut-back on the formatting issues.